Mu2eII_SM21: Update primary proton gun to work with Mu2e-II targets

michaelmackenzie opened the pull request:
Mu2eII_SM21:
Updating the primary proton gun implementation to work with Mu2e-II era targets.
Also added a few Mu2e-II specific configuration files.

Before this update, using the primary proton gun leads to an exception due to GeomHandle\<ProductionTarget\> not existing:
```
%MSG-s ArtException:  PostEndJob 19-Sep-2020 10:37:58 CDT ModuleEndJob
---- EventProcessorFailure BEGIN
  EventProcessor: an exception occurred during current event processing
  ---- ScheduleExecutionFailure BEGIN
    Path: ProcessingStopped.
    ---- GEOM BEGIN
      Failed to retrieve detector element of type N4mu2e16ProductionTargetE
      The above exception was thrown while processing module PrimaryProtonGun/generate run: 1
    ---- GEOM END
    Exception going through path trigmubeam
  ---- ScheduleExecutionFailure END
---- EventProcessorFailure END
%MSG
```

Validation run:
Compiles
JobConfig/validation/stoppedMuonsSingleStage.fcl runs (uses Mu2e era geometry and proton energy)
JobConfig/validation/stoppedMuonsSingleStage_Mu2eII.fcl runs (uses Conveyor Mu2e-II era target and proton energy)

FNALbuild commented:
Hi @michaelmackenzie,
You have proposed changes to files in these packages:
- ProductionTargetGeom
- JobConfig
- GeometryService
- EventGenerator
- Mu2eG4

which require these tests: build.

@Mu2e/write, @Mu2e/mu2eii_sm21_ci have access to CI actions on [Mu2eII_SM21](/Mu2e/Offline/tree/Mu2eII_SM21).

The following users requested to be notified about changes to these packages:
@resnegfk


[About FNALbuild.](https://mu2ewiki.fnal.gov/wiki/Git#GitHub_Pull_Request_Procedures_and_FNALbuild) [Code review on Mu2e/Offline.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#Code_Review)

sophiemiddleton commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for 5fab9b59200ed5915a72c3f1adaf53c4507d4c4f: build  (Build queue is empty)

michaelmackenzie commented:
For a reference point, the carbon conveyor target has a stopped muon rate of 14 / 100,000 = 1.40e-4 +- 0.37e-4 stops/POT.

sophiemiddleton commented:
That's quite low - right? Our current target is 0.0015 muons per POT....I think

FNALbuild commented:
:sunny: The tests passed at ref 5fab9b59200ed5915a72c3f1adaf53c4507d4c4f.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :heavy_check_mark: | Merged 5fab9b59200ed5915a72c3f1adaf53c4507d4c4f at 4d0c2032a7b2d7c03065810c866674eb206dc26f |
| scons build (prof) | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/scons.log). Build time: 11 min 56 sec |
| ceSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/ceSimReco.log) |
| g4test_03MT | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/g4test_03MT.log) |
| transportOnly | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/transportOnly.log) |
| PS | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/PS.log) |
| g4study | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/g4study.log) |
| cosmicSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/cosmicSimReco.log) |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (4) in 12 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 17 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//216/console).
Build artefacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.


Logfiles may also be accessed (at this link.)[https://gist.github.com/7d97281095bb8ba88e03372c3c30dbb1]

michaelmackenzie commented:
> That's quite low - right? Our current target is 0.0015 muons per POT....I think

I think it's roughly in agreement with naive estimates:
https://mu2eiiwiki.fnal.gov/wiki/Learn_about_Mu2e-II
but it should definitely be studied more.

sophiemiddleton commented:
Ah yes, that makes sense

oksuzian commented:
> > That's quite low - right? Our current target is 0.0015 muons per POT....I think
> 
> I think it's roughly in agreement with naive estimates of 1.9e-4 stops/POT:
> https://mu2eiiwiki.fnal.gov/wiki/Learn_about_Mu2e-II
> but it should definitely be studied more.

In the past, I was getting 1.3E-4 with the Mu2e pencil-like target.
I agree this should be studied and cross-checked with other generators (g4bl, MARS, FLUKA...)

Can we review and merge this PR?

michaelmackenzie commented:
I ran 10x the POT statistics, and got 87 stops / 1e6 POT = 0.87e-4 +- 0.09e-4 stops/POT (less than 1hr to run on mu2ebuild01 for reference). This is lower that hoped for, but could be due to the target design, the choice of carbon, or the alignment of the proton beam with the target (or many other factors).

I sent the origin and angle of the primary proton gun I used to Vitaly, which I'll also add here:
Gun origin = (3964.46,0.00575,-5915.45)
Gun rotation =
   [ (    0.970296             0      0.241922)
     (           0             1             0)
     (   -0.241922             0      0.970296) ]

Here is the origin and beam rotation for the gun with the Mu2e era target:
Gun origin = (3930.61,0,-6057.77)
Gun rotation =
   [ (    0.970296             0      0.241922)
     (           0             1             0)
     (   -0.241922             0      0.970296) ]

oksuzian commented:
> I ran 10x the POT statistics, and got 87 stops / 1e6 POT = 0.87e-4 +- 0.09e-4 stops/POT (less than 1hr to run on mu2ebuild01 for reference). This is lower that hoped for, but could be due to the target design, the choice of carbon, or the alignment of the proton beam with the target (or many other factors).
> 
> I sent the origin and angle of the primary proton gun I used to Vitaly, which I'll also add here:
> Gun origin = (3964.46,0.00575,-5915.45)
> Gun rotation =
> [ ( 0.970296 0 0.241922)
> ( 0 1 0)
> ( -0.241922 0 0.970296) ]
> 
> Here is the origin and beam rotation for the gun with the Mu2e era target:
> Gun origin = (3930.61,0,-6057.77)
> Gun rotation =
> [ ( 0.970296 0 0.241922)
> ( 0 1 0)
> ( -0.241922 0 0.970296) ]

How much do you get with Mu2e target: Hayman or pencil-like?
Would be nice to have these numbers for the reference.
What's the fcl file you use for this?

michaelmackenzie commented:
I got 76 stops / 50,000 POT = 0.00152 +- 0.00017 stops/POT for the current Mu2e target.
This agrees with Pasha's fairly recent calculation of 1.59e-3.
This is using JobConfig/validation/stoppedMuonsSingleStage.fcl using the geometry file:
Mu2eG4/geom/geom_2019_PhaseI_hayman_v2.txt (Hayman-v2.1 target+supports)

For the Mu2e-II rates, I used JobConfig/validation/stoppedMuonsSingleStage_Mu2eII.fcl using the geometry file:
Mu2eG4/geom/geom_Mu2eII_2020.txt (Conveyor target without supports)

oksuzian commented:
I meant to estimate the stopping rates using 800 MeV protons for:  Mu2e and Mu2e-II targets.
Did you use 8 GeV for the stopping rate estimate for the conveyor target?

michaelmackenzie commented:
I used the fcl file listed above, that uses a new Mu2e-II specific global constants file.
This global constants file just includes the default one, but changes the proton kinetic energy to 800 MeV.
https://github.com/Mu2e/Offline/pull/252/files?file-filters%5B%5D=.hh&file-filters%5B%5D=.txt#diff-0a8f247cc3d22a4c6c1794ee41b5dbb9

The other fcl file is unchanged (other than perhaps the geometry file), so using the 8000 MeV protons for the Mu2e era stopping rate.

oksuzian commented:
Can we adjust the code such that the following is used by all the fcl by default?
physics.producers.generate.mu2eii : true
services.GeometryService.inputFile : "JobConfig/common/geom_mu2eii.txt"
services.GlobalConstantsService.inputFile : "Mu2eG4/test/globalConstants_mu2eii.txt"

oksuzian commented:
It would great to compare Mu2e and Mu2e-II, using the same energies.
Otherwise, it's hard to make comparisons from these numbers.

brownd1978 commented:
Hi Yuri,
I thought the 2 MeV beam completely missed the Mu2e target.

Dave

On Tue, Sep 22, 2020 at 3:26 PM oksuzian <notifications@github.com> wrote:

> It would great to compare Mu2e and Mu2e-II, using the same energies.
> Otherwise, it's hard to make comparisons from these numbers.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/252#issuecomment-697014803>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH575FUBJHDQ7V4D4Y73LSHEQAJANCNFSM4RT4HGKA>
> .
>


-- 
David Nathan Brown            Dave_Brown@lbl.gov
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

oksuzian commented:
I believe we start right in front of the production target.
I ran this configuration in the past. I was observing 1.31E-4 stops/POT, which is higher than what we are getting (0.87E-4) with the conveyor target.

michaelmackenzie commented:
> Can we adjust the code such that the following is used by all the fcl by default?
> physics.producers.generate.mu2eii : true

This is PrimaryProtonGun specific, would you prefer the value to default to true in the code?
Or to update the prolog implementation with this line in EventGenerators/fcl/prolog.fcl?

> services.GeometryService.inputFile : "JobConfig/common/geom_mu2eii.txt"
> services.GlobalConstantsService.inputFile : "Mu2eG4/test/globalConstants_mu2eii.txt"

I could point JobConfig/common/geom_baseline.txt to the new geometry. I could also update the base services definitions to use this global constants file.

I had done it this way so old fcl behaved the same as before. I wasn't sure of the style that was preferred: update the defaults or leave them the same and make new fcl based on the originals but with the Mu2e-II specific components added.
I'm happy to update the defaults to be Mu2e-II specific!

michaelmackenzie commented:
> It would great to compare Mu2e and Mu2e-II, using the same energies.
> Otherwise, it's hard to make comparisons from these numbers.

Sure, I can try running the Hayman target without pbar absorbers and an 800 MeV beam for comparison.
This will change the shape (bend+balls vs straight+cylinders) and material, which could make it hard to know what the main affect is.

michaelmackenzie commented:
Running with the current Hayman v2.1 target+supports, no pbar windows, and 800 MeV I get 111 stops / 1e6 POT = 1.11e-4 +- 0.11e-4 stops/POT.
This is similar to 1.3e-4, and being lower than the pencil like target (without supports?) is consistent with past Hayman studies, correct? What was the uncertainty on 1.3e-4?

oksuzian commented:
> > Can we adjust the code such that the following is used by all the fcl by default?
> > physics.producers.generate.mu2eii : true
> 
> This is PrimaryProtonGun specific, would you prefer the value to default to true in the code?
> Or to update the prolog implementation with this line in EventGenerators/fcl/prolog.fcl?
I think prolog is a better option.

> 
> > services.GeometryService.inputFile : "JobConfig/common/geom_mu2eii.txt"
I think you have that in Mu2eG4/geom/geom_common_current.txt
#include "Mu2eG4/geom/geom_Mu2eII_2020.txt"

> > services.GlobalConstantsService.inputFile : "Mu2eG4/test/globalConstants_mu2eii.txt"
There are 160 fcl files that include the following line:
GlobalConstantsService : { inputFile : "Mu2eG4/test/globalConstants_01.txt" }
Should we include "Mu2eG4/test/globalConstants_mu2eii.txt" in Mu2eG4/test/globalConstants_01.txt?
Similar to what we do for the geometry file.

michaelmackenzie commented:
Odd that so many fcl file are defining the global constants file themselves!
I'm usually hesitant to change an older versioned file, but in this case it would just be changing 8000 -> 800 for now.
These changes may make it more confusing if we ever attempt to merge back into master though (which I think we should consider doing for many of these tools, for future Mu2e-II studies).

So I'll:
- Update the primary proton gun prolog definition to be Mu2e-II specific
- Update JobConfig/common/geom_baseline.txt to point to Mu2e-II geometry (should I also change Mu2eG4/geom/geom_common.txt?)
- Update Mu2eG4/test/globalConstants_01.txt to include the Mu2e-II specific file at the end, leaving us a clear place to add new updates to the constants

I can then also either update the validation script I added to be Mu2e era testing for comparisons or remove it since the original script will now be Mu2e-II era.

kutschke commented:
Hi Michael,


From: michaelmackenzie <notifications@github.com>
Reply-To: Mu2e/Offline <reply@reply.github.com>
Date: Wednesday, September 23, 2020 at 8:26 AM
To: Mu2e/Offline <Offline@noreply.github.com>
Cc: Robert Kutschke <kutschke@fnal.gov>, Team mention <team_mention@noreply.github.com>
Subject: Re: [Mu2e/Offline] Mu2eII_SM21: Update primary proton gun to work with Mu2e-II targets (#252)



Odd that so many fcl file are defining the global constants file themselves!



This is ancient history – most date from long before the definition in prologs of standard sets of services.  We are asking people to refactor to use the modern style as targets of opportunity.  If anyone is doing a major style update, I recommend that it be done in master and merged into Mu2eII_SM21.



I will be in meetings most of the day and can say more after the CompSoft meeting.

I'm usually hesitant to change an older versioned file, but in this case it would just be changing 8000 -> 800 for now.
These changes may make it more confusing if we ever attempt to merge back into master though (which I think we should consider doing for many of these tools, for future Mu2e-II studies).

So I'll:
·         Update the primary proton gun prolog definition to be Mu2e-II specific
·         Update JobConfig/common/geom_baseline.txt to point to Mu2e-II geometry (should I also change Mu2eG4/geom/geom_common.txt?)
·         Update Mu2eG4/test/globalConstants_01.txt to include the Mu2e-II specific file at the end, leaving us a clear place to add new updates to the constants

I can then also either update the validation script I added to be Mu2e era testing for comparisons or remove it since the original script will now be Mu2e-II era.

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_252-23issuecomment-2D697362667&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=AJ-G7CNgWeBgNw1tSkyivUBBnJQL84-vczLFrSwugag&s=3uGEL2tinFdSLCKwRTBn_XoSB61j2aSdjDlJnURv6aA&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZT27ULV2OAXF6AYYRTSHHZO7ANCNFSM4RT4HGKA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=AJ-G7CNgWeBgNw1tSkyivUBBnJQL84-vczLFrSwugag&s=Igtb3qtTbPt6zwTrOkVINlhVTwquOA4X_a977KzsuD4&e=>.

michaelmackenzie commented:
Hi Rob,

Do you have any opinions on the suggested changes or should I go forward with the edits as suggested?

Best,
Michael

sophiemiddleton commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for 5fab9b59200ed5915a72c3f1adaf53c4507d4c4f: build  (Build queue has 7 jobs)

FNALbuild commented:
:sunny: The tests passed at ref 5fab9b59200ed5915a72c3f1adaf53c4507d4c4f.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :heavy_check_mark: | Merged 5fab9b59200ed5915a72c3f1adaf53c4507d4c4f at 4d0c2032a7b2d7c03065810c866674eb206dc26f |
| scons build (prof) | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/scons.log). Build time: 15 min 31 sec |
| ceSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/ceSimReco.log) |
| g4test_03MT | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/g4test_03MT.log) |
| transportOnly | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/transportOnly.log) |
| PS | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/PS.log) |
| g4study | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/g4study.log) |
| cosmicSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/cosmicSimReco.log) |
| rootOverlaps | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/rootOverlaps.log) |
| g4surfaceCheck | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (4) in 12 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 17 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//233/console).
Build artefacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.


Logfiles may also be accessed [at this link.](https://gist.github.com/6e02c3da4cea90fbb1aa16f4959e0a9c)

michaelmackenzie commented:
Given there were no further comments on the suggestions, I went forward making the recommended changes.
I changed the default primary proton gun to expect a Mu2e-II era target, updated the proton KE, and changed the default geometry to Mu2e-II.
I also added a few configuration files that are for Mu2e-I, and an example script to show how to switch to them.

Validation:
Re-ran the validation stopped muon script with both Mu2e-II era and Mu2e-I era configurations.
These returned 1.40e-4 +- 0.37e-4 and 1.70e-3 +- 0.41e-3 muon stops / POT respectively, consistent with before.

sophiemiddleton commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for 2084e5803610f77e4b2c8d3d9ad4daee3861dc00: build  (Build queue has 3 jobs)

FNALbuild commented:
:umbrella: The tests failed for ref 2084e5803610f77e4b2c8d3d9ad4daee3861dc00.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :heavy_check_mark: | Merged 2084e5803610f77e4b2c8d3d9ad4daee3861dc00 at 4d0c2032a7b2d7c03065810c866674eb206dc26f |
| scons build (prof) | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/scons.log). Build time: 13 min 29 sec |
| ceSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/ceSimReco.log) |
| g4test_03MT | :x: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/g4test_03MT.log) |
| transportOnly | :x: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/transportOnly.log) |
| PS | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/PS.log) |
| g4study | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/g4study.log) |
| cosmicSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/cosmicSimReco.log) |
| rootOverlaps | :x: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/rootOverlaps.log) |
| g4surfaceCheck | :x: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (4) in 12 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 17 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//235/console).
Build artefacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.


Logfiles may also be accessed [at this link.](https://gist.github.com/4aae52948542e0a1d1eddcd155fae4f0)

michaelmackenzie commented:
It appears I accidentally mis-typed the configuration file name in geom_common.txt!
I'll double check the log files for other mistakes, then commit the fix.

kutschke commented:
Is this the explanation for the test failures just now?

On 9/30/20, 1:28 PM, "michaelmackenzie" <notifications@github.com<mailto:notifications@github.com>> wrote:

It appears I accidentally mis-typed the configuration file name in geom_common.txt!
I'll double check the log files for other mistakes, then commit the fix.

michaelmackenzie commented:
All of the errors pointed back to that typo, which should now be corrected.
Running Mu2eG4/fcl/transportOnly.fcl confirms it's been fixed, sorry!

kutschke commented:
No worries.  That’s why we have the CI system.

On 9/30/20, 1:33 PM, "michaelmackenzie" <notifications@github.com<mailto:notifications@github.com>> wrote:


All of the errors pointed back to that typo, which should now be corrected.
Running Mu2eG4/fcl/transportOnly.fcl confirms it's been fixed, sorry!

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_252-23issuecomment-2D701567720&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Vg81FS9wBF3iwcxJXhYACKDo-FAOPvw3-tWpb4jzraQ&s=ccJYTLnCwwUvGlRGqCr3xqy_BWjAm0Fv9aycvqtQxxw&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZSYBGMB7RIVMJ7JGFTSIN2V3ANCNFSM4RT4HGKA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Vg81FS9wBF3iwcxJXhYACKDo-FAOPvw3-tWpb4jzraQ&s=yKSqxkdf7hYUvSSGaUzRzPZ-zCU17fAljD1_KWFOaB0&e=>.

sophiemiddleton commented:
Ok, shall I rerun the test?

michaelmackenzie commented:
Sure, I think it should run cleanly now

sophiemiddleton commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for affd3ca752fea514956c0e6c92da752f4dd9f80e: build  (Build queue has 2 jobs)

FNALbuild commented:
:sunny: The tests passed at ref affd3ca752fea514956c0e6c92da752f4dd9f80e.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :heavy_check_mark: | Merged affd3ca752fea514956c0e6c92da752f4dd9f80e at 4d0c2032a7b2d7c03065810c866674eb206dc26f |
| scons build (prof) | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/scons.log). Build time: 12 min 05 sec |
| ceSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/ceSimReco.log) |
| g4test_03MT | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/g4test_03MT.log) |
| transportOnly | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/transportOnly.log) |
| PS | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/PS.log) |
| g4study | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/g4study.log) |
| cosmicSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/cosmicSimReco.log) |
| rootOverlaps | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/rootOverlaps.log) |
| g4surfaceCheck | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (4) in 12 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 17 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//236/console).
Build artefacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.


Logfiles may also be accessed [at this link.](https://gist.github.com/6ea6bdd704d2dc891dc98ae887b6d1f2)

kutschke commented:
Sophie reports by email that the review is complete.  Merging now.

FNALbuild commented:
:memo: The HEAD of `Mu2eII_SM21` has changed to 12d3e6713a26340f0d09bd24aa5f4f1e076059b1. Tests are now out of date.
